addon/components/mdc-drawer.js

```diff
@@ -0,0 +1,123 @@
+//region Ember Hooks
+  layout,
+  classNames: ['mdc-drawer'],
+  classNameBindings: ['mdcClassNames', 'isPermanent:mdc-permanent-drawer', 'isPersistent:mdc-persistent-drawer', 'isTemporary:mdc-temporary-drawer'],
+
+  init() {
+    this._super(...arguments);
+    set(this, 'changeHandlers', Ember.A([]));
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+    this.updateOpenness();
+  },
+  didUpdateAttrs() {
+    this._super(...arguments);
+    this.updateOpenness();
+  },
+  //endregion
+
+  //region Attributes
+  /**
+   * @type {Boolean}
+   */
+  permanent: true,
+  /**
+   * @type {Boolean}
+   */
+  persistent: false,
+  /**
+   * @type {Boolean}
+   */
+  temporary: false,
+  /**
+   * @type {Boolean}
+   */
+  open: false,
+  onopen: x => x,
+  onclose: x => x,
+  //endregion
+
+  //region ComputedProperties
+  isPermanent: computed('permanent', 'persistent', 'temporary', function () {
+    const {permanent, persistent, temporary} = getProperties(this, 'permanent', 'persistent', 'temporary');
+    if (permanent && (temporary || persistent)) {
+      throw new Ember.Error('Cannot be permanent and temporary or persistent');
+    }
+    return permanent;
+  }),
+  isPersistent: computed('permanent', 'persistent', 'temporary', function () {
+    const {permanent, persistent, temporary} = getProperties(this, 'permanent', 'persistent', 'temporary');
+    if (persistent && (permanent || temporary)) {
+      throw new Ember.Error('Cannot be persistent and permanent or temporary');
+    }
+    return persistent;
+  }),
+  isTemporary: computed('permanent', 'persistent', 'temporary', function () {
+    const {permanent, persistent, temporary} = getProperties(this, 'permanent', 'persistent', 'temporary');
+    if (temporary && (permanent || persistent)) {
+      throw new Ember.Error('Cannot be temporary and permanent or persistent');
+    }
+    return temporary;
+  }),
+  //endregion
+
+  //region Methods
+  updateOpenness() {
+    const foundation = get(this, 'foundation');
+    if (!foundation) { return; }
+    const open = get(this, 'open');
+    if (foundation.isOpen() && !open) {
+      foundation.close();
+    }
+    if (!foundation.isOpen() && open) {
+      foundation.open();
+    }
+  },
+  /**
+   * @returns {MDCPersistentDrawerFoundation|MDCTemporaryDrawerFoundation}
+   */
+  createFoundation() {
+    const {permanent, persistent} = getProperties(this, 'permanent', 'persistent');
+    if (permanent) {
+      return { init() {}, destroy() {} };
+    }
+    const Foundation = persistent ? MDCPersistentDrawerFoundation : MDCTemporaryDrawerFoundation;
+    const {FOCUSABLE_ELEMENTS, ITEMS_SELECTOR, DRAWER_SELECTOR} = Foundation.strings;
+    const adapter = {
+      addClass: className => get(this, 'mdcClasses').addObject(className),
+      removeClass: className => get(this, 'mdcClasses').removeObject(className),
+      hasClass: className => get(this, 'element.classList').contains(className),
```

Comment:
I wonder if you'd be better off here using `mdcClasses` as a source of truth and writing this as such:

```js
className => get(this, 'mdcClasses').includes(className)
``